File: android/src/main/kotlin/ru/kardenvan/flutter_yandex_ads_plus/platform_api/ad_event/AdEvent.kt
package ru.kardenvan.flutter_yandex_ads_plus.platform_api.ad_event

data class AdEvent(
    private val uid: String,
    private val type: String,
    private val parameters: Map<String, Any?>? = null,
) {
    fun toMap(): Map<*, *> {
        return mapOf(
            "uid" to uid,
            "type" to type,
            "parameters" to parameters,
        )
    }
}

File: ios/Classes/platform_api/ad_event/AdEvent.swift
//
//  AdEvent.swift
//  flutter_yandex_ads_plus
//
//  Created by apple on 30.12.2022.
//

import Foundation

class AdEvent {
    private let uid: String
    private let type: String
    private let parameters: [String: Any?]?

    init(
        uid: String,
        type: String,
        parameters: [String: Any?]? = nil
    ) {
        self.uid = uid
        self.type = type
        self.parameters = parameters
    }
    
    func toMap() -> [String: Any?] {
        return [
            "uid": uid,
            "type": type,
            "parameters": parameters
        ]
    }
}
